ddtrace/internal/core.py

```diff
@@ -40,49 +39,72 @@ def has_listeners(self, event_id):
 
     def on(self, event_id, callback):
         # type: (str, Callable) -> None
-        with self._dispatch_lock:
+        if callback not in self._listeners[event_id]:
             self._listeners[event_id].append(callback)
 
     def reset(self):
-        with self._dispatch_lock:
-            self._listeners = defaultdict(list)
+        if hasattr(self, "_listeners"):
+            del self._listeners
```

Comment:
Why using `del` here ?

Reply:
I wanted to be *really sure* that `_listeners` isn't persisting across `reset()` calls. It's overkill 😅